### PR TITLE
Use "foreach" for 3D imrotate, document it

### PR DESCRIPTION
**Pull request by JeffFessler**
Also use `RotateMode` to make it extendable and easier for dispatch.

### CODE REVIEW

File: src/rotatez.jl [hunk 1]
```diff
@@ -506,3 +498,78 @@ function imrotate_adj!(output::AbstractMatrix{<:RealU},
...
+
+    fun = z -> imrotate_adj!(
+        (@view output[:,:,z]),
+        (@view image3[:,:,z]),
+        θ,
+        plans[Threads.threadid()],
+    )
+
+    ntasks = length(plans)
+    Threads.foreach(fun, _setup(1:size(image3,3)); ntasks)
```
Comment (ZongyuLi-umich) [inline src/rotatez.jl:486]: I see, so we might want to use `foreach` everywhere?

File: test/adjoint-rotate.jl [hunk 2]
```diff
@@ -25,32 +24,27 @@ end
         image3 = rand(T, nx, nx, 4)
         result = similar(image3)
         imrotate!(result, image3, π/6, plan)
-        @test maximum(result) ≤ 1.1
-        imrotate_adj!(result, image3, -π/6, plan)
-        @test maximum(result) ≤ 1.1
+        @test maximum(result) ≤ 1.01
+        imrotate_adj!(result, image3, π/6, plan)
+        @test maximum(result) ≤ 1.5
```
Comment (ZongyuLi-umich) [inline test/adjoint-rotate.jl:29]: that's something I have not figured out yet, why the 2d interpolation adjoint enlarge the range of pixels so much?

File: src/rotatez.jl [hunk 3]
```diff
@@ -20,7 +16,7 @@ function linearinterp!(
     A::SparseInterpolator{<:AbstractFloat},
     x::AbstractVector{<:RealU},
 )
-    # x must be a constant vector
+    # x must be a constant vector todo: why? it only uses the 1st and end of x!?
```
[the remark this answers is not in the document]
Comment (JeffFessler) [reply src/rotatez.jl:19]: still mysterious that it uses only x[1] and x[end] !?

File: src/rotatez.jl (hunk 1 above)
Comment (JeffFessler) [reply src/rotatez.jl:486]: Probably - after we do timing tests of this...

**Comment by ZongyuLi-umich** [comment]
Done. Please check out the new pull request.

**Pull request merged**